# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/openjdk/hppa.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- jdk-jdk-23-ga/make/autoconf/platform.m4.vanilla	2024-10-17 20:49:30.619900849 +0200
+++ jdk-jdk-23-ga/make/autoconf/platform.m4	2024-10-17 20:53:20.896564265 +0200
@@ -66,6 +66,12 @@
       VAR_CPU_BITS=64
       VAR_CPU_ENDIAN=little
       ;;
+    hppa)
+      VAR_CPU=hppa
+      VAR_CPU_ARCH=parisc
+      VAR_CPU_BITS=32
+      VAR_CPU_ENDIAN=big
+      ;;
     ia64)
       VAR_CPU=ia64
       VAR_CPU_ARCH=ia64
--- jdk-jdk-23-ga/src/hotspot/os/linux/os_linux.cpp.vanilla	2024-11-17 16:10:58.663509765 +0100
+++ jdk-jdk-23-ga/src/hotspot/os/linux/os_linux.cpp	2024-11-17 16:11:51.306844055 +0100
@@ -1789,7 +1789,7 @@
   static  Elf32_Half running_arch_code=EM_ALPHA;
 #elif  (defined MIPSEL)
   static  Elf32_Half running_arch_code=EM_MIPS_RS3_LE;
-#elif  (defined PARISC)
+#elif  (defined PARISC) || (defined HPPA)
   static  Elf32_Half running_arch_code=EM_PARISC;
 #elif  (defined MIPS)
   static  Elf32_Half running_arch_code=EM_MIPS;
